pandas/tests/indexes/test_setops.py

```diff
@@ -39,14 +28,14 @@ def test_union_same_types(indices):
     assert idx1.union(idx2).dtype == idx1.dtype
 
 
-def test_union_different_types(index_pair):
+def test_union_different_types(indices, indices2):
```

Comment:
for a future reader, could add a comment along the lines that the test runs for all permutations of index types, but that this test only considers combinations.